bin/node/cli/src/service.rs

```diff
@@ -102,14 +105,17 @@ pub fn new_partial(config: &Configuration) -> Result<sc_service::PartialComponen
 
 	let import_setup = (block_import, grandpa_link, babe_link);
 
+
 	let (rpc_extensions_builder, rpc_setup) = {
 		let (_, grandpa_link, babe_link) = &import_setup;
 
 		let justification_stream = grandpa_link.justification_stream();
 		let shared_authority_set = grandpa_link.shared_authority_set().clone();
 		let shared_voter_state = grandpa::SharedVoterState::empty();
+		let finality_proof_provider =
+			GrandpaFinalityProofProvider::new_for_service(backend.clone(), client.clone());
```

Comment:
My understanding is that it should be fine to move the creation of the finality proof provider from `new_full_base` to `new_partial` ... ?